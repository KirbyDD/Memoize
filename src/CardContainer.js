import React, { Component } from "react";
import Card from "./Card.js"
import "./CardContainer.scss"

class CardContainer extends Component {
  constructor(props) {
    super(props);
    this.state= { 
      questionNum: 1
    }
  }

  handleNext = (e) => {
    this.setState({questionNum: this.state.questionNum+1})
  }

  handlePrev = (e) => {
    this.setState({questionNum: this.state.questionNum-1})
  }

  render(){
    return (
      <main> 
        {this.props.questions
        .filter(question => question.id === this.state.questionNum)
        .map(question => {
          return <Card handleNext={this.handleNext}
                       handlePrev={this.handlePrev}
                       question={question.question}
                       answer={question.answer}
                       key={question.id}
                       image={question.image}
                       />
          })
        }
      </main>
    )
  }
}

export default CardContainer;