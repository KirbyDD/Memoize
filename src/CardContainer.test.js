import React from 'react';
import CardContainer from './CardContainer.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const mockQuestions = [
  {
  id: 1,
  question: "Which of the following is an ES5 variable declaration: var, let, or const?",
  image: null,
  answer: "var"
  },
  {
  id: 2,
  question: "Which of the following is a non-mutable variable declaration: var, let, or const?",
  image: null,
  answer: "const"
  },
  {
  id: 3,
  question: "Which of the following is a mutable ES6 variable declaration: var, let, or const?",
  image: null,
  answer: "let"
  }
]




describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CardContainer questions={mockQuestions}/>
    )
  })

  it('should match snapshot when data is passed in ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have state with default value', () => {
    expect(wrapper.state()).toEqual({ questionNum: 1 });
  })

  it('should increment questionNum up one', () => {
    wrapper.instance().handleNext();
    expect(wrapper.state()).toEqual({ questionNum: 2 })
  })

  it('should decrement questionNum down one', () => {
    wrapper.instance().handlePrev();
    expect(wrapper.state()).toEqual({ questionNum: 0 })
  })
})