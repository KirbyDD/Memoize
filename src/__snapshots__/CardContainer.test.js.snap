// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when data is passed in  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    questions={
      Array [
        Object {
          "answer": "var",
          "id": 1,
          "image": null,
          "question": "Which of the following is an ES5 variable declaration: var, let, or const?",
        },
        Object {
          "answer": "const",
          "id": 2,
          "image": null,
          "question": "Which of the following is a non-mutable variable declaration: var, let, or const?",
        },
        Object {
          "answer": "let",
          "id": 3,
          "image": null,
          "question": "Which of the following is a mutable ES6 variable declaration: var, let, or const?",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          answer="var"
          handleNext={[Function]}
          handlePrev={[Function]}
          image={null}
          question="Which of the following is an ES5 variable declaration: var, let, or const?"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "answer": "var",
          "handleNext": [Function],
          "handlePrev": [Function],
          "image": null,
          "question": "Which of the following is an ES5 variable declaration: var, let, or const?",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            answer="var"
            handleNext={[Function]}
            handlePrev={[Function]}
            image={null}
            question="Which of the following is an ES5 variable declaration: var, let, or const?"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "answer": "var",
            "handleNext": [Function],
            "handlePrev": [Function],
            "image": null,
            "question": "Which of the following is an ES5 variable declaration: var, let, or const?",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
